### PR TITLE
Upgrade `request` to `axios`

### LINKED ISSUE #240: Replace the deprecated and vulnerable dependency package `request`
The direct dependency package `request` has been deprecated in Feb 2020 (https://www.npmjs.com/package/request). All versions of `request` including the latest one are affected by prototype pollution vulnerability (https://sca.analysiscenter.veracode.com/vulnerability-database/security/sca/vulnerability/sid-21913/summary)

Maintainers of the package have composed the list of alternative libraries for replacement: https://github.com/request/request/issues/3143

### PR DESCRIPTION
**Pull request by sameerag**
This PR updates `request` to `axios` for all `get` and `post` methods in the library.

Pending:
One unit test is failing where axios seems to make an extra NW call in nock for username-password.ts. All other cases pass.

### CODE REVIEW

**Review by jasonnutter — CHANGES_REQUESTED** [review]
It doesn't look like an updated `package.json` is included.

**Review by hectormmg — APPROVED** [review]

**Comment by sameerag** [comment]
> It doesn't look like an updated `package.json` is included.

Updated now. Please review.

File: package.json [hunk 1]
```diff
@@ -34,7 +34,8 @@
     "underscore": ">= 1.3.1",
     "uuid": "^3.1.0",
     "xmldom": ">= 0.1.x",
-    "xpath.js": "~1.1.0"
+    "xpath.js": "~1.1.0",
+    "axios": "^0.21.1"
```
Comment (jasonnutter) [inline package.json:37]: Be sure to remove `request` as well.

Comment (sameerag) [reply package.json:37]: I kept it to fix the last failing unit test, request is making one less mock nw call than axios. Will change soon after I fix it.

**Comment by gijsmin** [comment]
Are there any updates on when this PR will be merged?

File: package.json (hunk 1 above)
Comment (sameerag) [reply package.json:37]: Can you review now and approve?

**Comment by sameerag** [comment]
@Gm33 yes, we plan to merge it this week and planning a release by Feb 1st. Also please give us early feedback on the PR if you guys want to take it for a spin testing.

**Review by jasonnutter — APPROVED** [review]

**Comment by vinu-vrize** [comment]
@jasonnutter @sameerag when would be the next release planned with this particular issue fixed.

**Comment by gijsmin** [comment]
@sameerag, when do you expect to release the last version with the migration to axios?

**Comment by sameerag** [comment]
A beta version is released yesterday [here](https://www.npmjs.com/package/adal-node/v/0.2.2-beta.0). We plan to wait a week for feedback from customers and release mainline. Please test  with the beta with `npm install adal-node@beta` and let us know if you have any feedback.

cc @hectormmg

**Pull request merged**